### DISCUSSION TITLE
Why exoplayer over VLC ?

### DISCUSSION THREAD

**Discussion by AntoineBarj** [jellyfin/jellyfin-androidtv#4049] [General]
Hello there .

My Jellyfinserver is set up on a docker image on my NAs, that has poor transcoding performances.
But as I run a Jellyfish client on my Nvidia shield, I did not expect any need of transcoding.

Indeed, any video file can be played flawless on the Android TV VLC app (that I currently use as external player from the jellyfin client). HDR, 4k, various subs and files types . 
From a newbie like me, I wondered what could be the strategical reasons that lead to use exoplauer lib I et the VLC one. I naively think that the flawless experience on the VLC app should be the same on jellyfin ;). Right now, the playback is  natively struggling for 80% of my mkv files, with either HDR or subs (or both).

Tl,DR : what is the reason behind choosing the exoolayer library over the VLC one ?

BTW :  congrats to the Dev for this maintained project !!

**Comment by hendrik1120** [comment]
Security issues and a pain to work with, also no stable release since 2022
See:
https://github.com/jellyfin/jellyfin-androidtv/pull/2534
https://github.com/jellyfin/jellyfin-androidtv/pull/3639

**Comment by BenMcLean** [comment]
Another reason to prefer exoplayer over VLC is that exoplayer supports audio passthrough for surround sound while VLC on Android only does stereo. Note I said on Android. VLC for PC handles surround sound audio fine while, in my experience, VLC for Android never does.
